src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/UseSpanClearInsteadOfFill.cs

```diff
@@ -0,0 +1,149 @@
+                    {
+                        context.ReportDiagnostic(invocation.CreateDiagnostic(s_Rule));
+                    }
+                },
+                OperationKind.Invocation);
+        }
+
+        private static bool IsDefaultValue(IArgumentOperation argumentOperation)
+        {
+            var value = argumentOperation.Value;
+            var type = argumentOperation.Parameter.Type;
+            var constantOpt = value.ConstantValue;
+
+            if (constantOpt.HasValue)
+            {
+                if (constantOpt.Value == null)
+                {
+                    // null must be default value for any valid type
+                    return true;
+                }
+
+                if (argumentOperation.Type.IsNullableValueType())
+                {
+                    // 0 isn't default value for T?
+                    return false;
+                }
+
+                // enum/nint are treated as integers
+                // This is missing default value of DateTime literal for VB,
+                // but since VB doesn't properly support Span, just don't consider it
+                switch (constantOpt.Value)
+                {
+#pragma warning disable IDE0004 // Remove Unnecessary Cast - false positive
+                    case (byte)0 or (short)0 or (int)0 or (long)0 or
+                        (sbyte)0 or (ushort)0 or (uint)0 or (ulong)0 or
+                        false or '\0':
+                        return true;
+#pragma warning restore IDE0004 // Remove Unnecessary Cast
+
+                    // -0 is not all bits zero. Handle them by bits
+                    case float f:
+                        // SingleToInt32Bits not available in netstandard2.0
+                        return BitConverter.DoubleToInt64Bits(f) == 0;
+
+                    case double d:
+                        return BitConverter.DoubleToInt64Bits(d) == 0;
+
+                    case decimal d:
+                        return decimal.GetBits(d).All(b => b == 0);
+
+                    default:
+                        return false;
+                }
+            }
+
+            if (value is IDefaultValueOperation)
+            {
+                return SymbolEqualityComparer.Default.Equals(value.Type, type);
+            }
+
+            if (value is IConversionOperation { Operand: IDefaultValueOperation defaultValue })
+            {
+                // handle the conversion of default literal only
+                return SymbolEqualityComparer.Default.Equals(defaultValue.Type, type);
+            }
+
+            if (value is IObjectCreationOperation objectCreation)
+            {
+                return objectCreation.Type.IsValueType
+                    && objectCreation.Arguments.IsEmpty
+                    && (objectCreation.Initializer?.Initializers.IsEmpty ?? true)
+                    && SymbolEqualityComparer.Default.Equals(value.Type, type);
+            }
```

Comment:
If I understand this correctly. I think this doesn't play nice with struct parameterless constructors feature?

```csharp
using System;

C c = default;
C c2 = new C();

Console.WriteLine(c.X); // 0
Console.WriteLine(c2.X); // 5

public struct C
{
    public int X;
    public C() => X = 5;
}
```

Reply:
Yes. I wrote the code when struct parameterless constructor was not a thing. It may be easier to test when ReferenceAssemblies.Net60 comes?

Reply:
@huoyaoyuan I think it's not related to .NET 6 RefenceAssemblies. Just a compiler version that has the feature is enough. If there is an available version already, it should be just upgrading `MicrosoftCodeAnalysisVersionForTests` to that version.

Also Net60 was actually added in https://github.com/dotnet/roslyn-sdk/commit/de9af89015295d30088cf95477c9079eb3c3d1b4, but not sure if a version is out or not.

Reply:
The feature was merged in roslyn by 17.0p3, however this repo is still referencing 4.0.0-2 . So I can't test it right now.

Reply:
@huoyaoyuan Did you try upgrading to `4.0.0-3.final` and encountered any issues?

Reply:
It should be fine, but I think the version should be updated by a separated PR like #5274 . /cc @sharwell

Reply:
The test version has been updated now.